eng/common/TestResources/New-TestResources.ps1

```diff
@@ -279,46 +279,13 @@ try {
         if (!$TenantId) {
             $TenantId = $context.Subscription.TenantId
         }
-
-        # If no test application ID is specified during an interactive session, create a new service principal.
-        if (!$TestApplicationId) {
```

Comment:
Blocking as I need time to digest this. All the SP handling - the brunt of what this script does to bootstrap deployment - was heavily tested - and modified and tested again and again - and should not be changed without considerable test cases including in different subs in different environments with both the remembered variable defined and not defined, and more.

Reply:
Happy to add test coverage;  there didn't seem to be a matrix or any resources related to doing so.  I've run tests against the Playground subscription and my personal subscription for the following scenarios:

 - Playground, no provisioner,  no test app
 - Playground, no provisioner, existing test app
 - Playground provisioner, no test app
 - Playground provisioner, existing test app
 - Personal, no provisioner,  no test app
 - Personal, no provisioner, existing test app
 - Personal, provisioner, no test app
 - Personal, provisioner, existing test app

I've also included the scenario where the provisioner and existing test app are the same principal.    Feel free to add additional scenarios or point me towards docs that I may have overlooked.

Reply:
And what about your command line arguments? Supplying the privisioner is currently not required, but your changes will make it so. It was to be automatic for end users but the CI supplies it's own.

Reply:
Why would end users need to supply a provisioner?  When they execute the script, it is running in the context of their permissions.  If they have rights to create a test application and grant it  `Owner` on the subscription, they have rights to create a resource group and run an ARM deployment.

I may be overlooking something, but I don't see why it would be necessary or desirable to log into the test application principal to run the deployment.  In most cases, that SP is running with less permissions than the interactive user due to the lack of ability to query AAD via the Graph API.

Reply:
Discussed offline.